pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/JavaQualifiedName.java

```diff
@@ -4,379 +4,156 @@
 
 package net.sourceforge.pmd.lang.java.ast;
```

Comment:
this class should probably not be part of this package

Reply:
Moving it out would be API breaking

Reply:
True, I've added it to #881. I'd avoid having the factory on this package non the less to avoid that breaking change in 7.0.0

Reply:
The next PR moves ~80% of the factory out into the visitor. If you're ok with it, I'd rather move it out of the package then, otherwise I'd have some merge conflicts to resolve ^^

Reply:
If the next PR doesn't make it into 6.2.0 we would be in the very difficult position of either having to roll this back before the release, or deprecate it in 6.3.0. Not the happiest path in either case... I'm trusting you on this one, but let's be careful.

Reply:
I'll open the next PR asap